sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/CSRClientProvider.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.kafka;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import java.io.Serializable;
+
+/**
+ * Provides instances of Confluent Schema Registry clients.
+ *
+ * <p>Please note, that any instance of {@link CSRClientProvider} must be {@link Serializable} to
+ * ensure it can be sent to worker machines.
+ */
+public interface CSRClientProvider extends Serializable {
+
+  SchemaRegistryClient getCSRClient();
+
+  String getSchemaRegistryUrl();
```

Comment:
We can't remove it since `SchemaRegistryClient` doesn't return `schemaRegistryUrl`, do it?

Reply:
`registryUrl` is an input parameter, not an output one. The user already knows the URL because he is building the client so that side is covered, and KafkaIO does not care about the `registryUrl` because it already has the built client. Or am I missing something?

Reply:
We need to set Kafka config option "schema.registry.url" with this value and we do this in `withCSRClientProvider(CSRClientProvider)`. Unfortunately, `SchemaRegistryClient` doesn't have an API method to return `schemaRegistryUrl`, so we use the one stored in `CSRClientProvider`. What is an issue to have `String getSchemaRegistryUrl()` there?

Reply:
Oh I see my mistake, sorry. Ugly `Map based APIs`. Ok then, but to the history I want to say that this is information leaking ! :stuck_out_tongue: